src/chromo/common.py

```diff
@@ -733,8 +739,8 @@ def cross_section(self, kin=None):
                     kin3.p2 = component
                     # this calls cross_section recursively, which is fine
                     cs = self.cross_section(kin3)
-                    for i, val in enumerate(dataclasses.astuple(cs)):
-                        cross_section[i] += fraction * val
+                    for field in dataclasses.fields(cs):
+                        cross_section[field.name] += fraction * cs[field.name]
```

Comment:
Sorry, I don't understand what the problem is with dataclasses.astuple. How is this a better solution?

Reply:
I hope we can converge on the solution I describe above, which adds a private method `CrossSectionData._mul_add`. The code here would change to `cross_section._mul_radd(fraction, cs)`

Reply:
There is no problem with dataclasses.astuple. I would be happy not touch the previous implementation, but the line:

```
cross_section[i] += fraction * val
```

gives an error: `'CrossSectionData' object is not subscriptable`.

The only way I found how to iterate over fields is to use getattr.

Reply:
I changed to use `cross_section._mul_radd(fraction, cs)`.